app/science_module/src/main.cpp

```diff
@@ -0,0 +1,189 @@
+/**
+  ******************************************************************************
+  * @file    main.c
+  * @author  Kevin Huo
+  * @version V1.0
+  * @date    13-May-2018
+  * @brief   Main function for science module.
+  *
+  *
+  * This file serves as the driver for the science module. It receives data
+  * from the CAN bus and generates an appropriate PWM signal for each module.
+  * Additionally, it reads sensor data and transmits through CAN when the 
+  * appropriate message is received.
+  ******************************************************************************
+*/
+#include "mbed.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "canlib.h"
+#include "pins.h"
+
+#define NUM_MODULES         4
+#define INIT_VECTOR_SIZE    5
+
+Serial pc(USBTX, USBRX);
+I2C i2c(I2C_SDA, I2C_SCL);
+
+typedef enum module_ids{
+    science_module_pwm_filter   = 400,
+    elevator_pwm_filter_ID      = 420,
+    drill_pwm_filter_ID         = 421,
+    temp_pwm_filter_ID          = 422,
+    flap_pwm_filter_ID          = 423,
+
+    elevator_pwm_ID             = 424,
+    drill_pwm_ID                = 425,
+    temp_pwm_ID                 = 426,
+    flap_pwm_ID                 = 427,
```

Comment:
what are these *_pwm_ID values for? they aren't used again, and these IDs aren't defined in the CAN spec